cardano-cli/src/Cardano/CLI/EraBased/Script/Proposal/Read.hs

```diff
@@ -0,0 +1,120 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE GADTs #-}
+
+module Cardano.CLI.EraBased.Script.Proposal.Read
+  ( readProposalScriptWitness
+  )
+where
+
+import           Cardano.Api
+import           Cardano.Api.Shelley
+
+import           Cardano.CLI.EraBased.Script.Proposal.Types
+import           Cardano.CLI.EraBased.Script.Read.Common
+import           Cardano.CLI.EraBased.Script.Types
+import           Cardano.CLI.Types.Common
+
+readProposalScriptWitness
+  :: MonadIOTransError (FileError CliScriptWitnessError) t m
+  => ConwayEraOnwards era
+  -> (ProposalFile In, Maybe CliProposalScriptRequirements)
+  -> t m (Proposal era, Maybe (ProposalScriptWitness era))
+readProposalScriptWitness w (propFp, Nothing) = do
+  proposal <-
+    conwayEraOnwardsConstraints w $
+      modifyError (fmap TextEnvelopeError) $
+        hoistIOEither $
+          readFileTextEnvelope AsProposal propFp
+  return (proposal, Nothing)
+readProposalScriptWitness w (propFp, Just certScriptReq) = do
+  let sbe = convert w
+  proposal <-
+    conwayEraOnwardsConstraints w $
+      modifyError (fmap TextEnvelopeError) $
+        hoistIOEither $
+          readFileTextEnvelope AsProposal propFp
```

Comment:
```suggestion
  (proposal, _) <- readProposalScriptWitness w (propFp, Nothing)
```

Reply:
This is more terse but I find it strange as we're saying there is no certificate script witness but there is.